compiler/optimizer/OMRLocalCSE.cpp

```diff
@@ -1355,7 +1355,8 @@ TR::Node* OMR::LocalCSE::getAvailableExpression(TR::Node *parent, TR::Node *node
          }
       }
 
-   if ((node->getOpCode().isArrayRef()) &&
+   if (node->hasPinningArrayPointer() &&
```

Comment:
Maybe you considered wrapping the hasPinningArrayPointer && computeIsInternalPointer tests inside a query routine so that you do not have the 2 tests in several places. It's not a big deal though.